client/app/dedicatedCloud/datacenter/drp/steps/common/first/dedicatedCloud-datacenter-drp-firstStep.controller.js

```diff
@@ -0,0 +1,76 @@
+export default class {
+  /* @ngInject */
+  constructor(
+    $q, $state, $stateParams, $translate, $uibModal,
+    Alerter, DedicatedCloudDrp, OvhApiDedicatedCloud,
+    DEDICATEDCLOUD_DATACENTER_DRP_IP_USAGE_MAC_ADDRESS_REG_EXP,
+    DEDICATEDCLOUD_DATACENTER_DRP_UNAVAILABLE_IP_STATUS,
+  ) {
+    this.$q = $q;
+    this.$state = $state;
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.$uibModal = $uibModal;
+    this.Alerter = Alerter;
+    this.DedicatedCloudDrp = DedicatedCloudDrp;
+    this.OvhApiDedicatedCloud = OvhApiDedicatedCloud;
+    this.MAC_ADDRESS_REG_EXP = DEDICATEDCLOUD_DATACENTER_DRP_IP_USAGE_MAC_ADDRESS_REG_EXP;
+    this.UNAVAILABLE_IP_STATUSES = DEDICATEDCLOUD_DATACENTER_DRP_UNAVAILABLE_IP_STATUS;
+  }
+
+  $onInit() {
+    this.loading = true;
+    this.drpInformations = this.$stateParams.drpInformations;
+
+    return this.$q.all({
+      ipAddressDetails:
+        this.DedicatedCloudDrp.getPccIpAddressesDetails(this.$stateParams.productId),
+      pcc: this.OvhApiDedicatedCloud.v6().get({
+        serviceName: this.$stateParams.productId,
+      }),
+    })
+      .then(({ ipAddressDetails, pcc }) => {
+        this.drpInformations.primaryPcc = pcc;
+        this.availableDatacenters = this.datacenters;
+        this.availableIpAddress = this.getAvailableAddresses(ipAddressDetails);
+
+        this.drpInformations.primaryDatacenter = this.availableDatacenters
+          .find(({ id }) => parseInt(this.$stateParams.datacenterId, 10) === id);
+        this.selectedIpAddress = this.availableIpAddress
+          .find(({ ip }) => ip === this.$stateParams.drpInformations.primaryEndpointIp);
+      })
+      .catch((error) => {
+        this.Alerter.error(`${this.$translate.instant('dedicatedCloud_datacenter_drp_get_state_error')} ${_.get(error, 'data.message', error)}`, 'dedicatedCloudDatacenterDrpAlert');
+      })
+      .finally(() => {
+        this.loading = false;
+      });
+  }
+
+  openModalOrderIpBlock() {
+    this.$uibModal.open({
+      templateUrl: 'ip/ip/order/ip-ip-order.html',
+      controller: 'IpOrderCtrl',
+      controllerAs: '$ctrl',
+    }).result
+      .then(() => this.DedicatedCloudDrp.getPccIpAddressesDetails(this.$stateParams.productId))
+      .then((ipAddressDetails) => {
+        this.availableIpAddress = this.getAvailableAddresses(ipAddressDetails);
+      });
+  }
+
+  getAvailableAddresses(ipAddressDetails) {
+    return ipAddressDetails
+      .filter(({ usageDetails }) => _.isNull(usageDetails)
+        && !this.UNAVAILABLE_IP_STATUSES.includes(usageDetails)
+        && !this.MAC_ADDRESS_REG_EXP.test(usageDetails));
+  }
+
+  goToPreviousStep() {
+    this.$state.go('app.dedicatedClouds.datacenter.drp', { selectedDrpType: this.drpInformations.drpType });
+  }
+
+  goToNextStep() {
+    this.$state.go(`app.dedicatedClouds.datacenter.drp.${this.drpInformations.drpType}.secondStep`, { drpInformations: this.drpInformations });
```

Comment:
Is your method need to return this Promise?

Reply:
No it doesn't